source/iterable.js

```diff
@@ -14,16 +16,7 @@ export const lineIterator = async function * (subprocess, {state}, streamName) {
 			return;
 		}
 
-		let buffer = '';
-		for await (const chunk of stream.iterator({destroyOnReturn: false})) {
-			const lines = `${buffer}${chunk}`.split(/\r?\n/);
-			buffer = lines.pop(); // Keep last line in buffer as it may not be complete
-			yield * lines;
-		}
-
-		if (buffer) {
-			yield buffer; // Yield any remaining data as the last line
-		}
+		yield * readline.createInterface({input: stream});
```

Comment:
We could remove a few more bytes by doing:

```js
		const {[streamName]: input} = await subprocess.nodeChildProcess;
		if (!input) {
		  yield * readline.createInterface({input});
		}
```

Reply:
I like early return.

Reply:
Me too

Reply:
:+1: